src/api/movies-api.ts

```diff
@@ -20,10 +22,10 @@ const getAllMovies = (): Promise<MovieEntity[]> => {
 
 
 const getAllMoviesByGenre = (genre : string ): Promise<MovieEntity[]> => {
```

Comment:
Here we can create a separate case, instead of two methods, let's have one

```typescript
interface MoviesFilter {
  pageIndex,
  pageSize,
   filter // object
}

const getMovies = (filter: MoviesFilter = createDefaultMoviesFilter()) {
}
```